packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/views/explorer/services/ExplorerNavigationService.java

```diff
@@ -86,4 +89,25 @@ private String getItemId(Object object) {
         }
         return result;
     }
+
+    private Optional<Object> getTreeItemObject(IEditingContext editingContext, String id, Tree tree) {
+        var optionalTreeDescription = this.getTreeDescription(editingContext, tree);
+
+        if (optionalTreeDescription.isPresent()) {
+            var variableManager = new VariableManager();
+            variableManager.put(IEditingContext.EDITING_CONTEXT, editingContext);
+            variableManager.put(TreeDescription.ID, id);
+            return Optional.ofNullable(optionalTreeDescription.get().getTreeItemObjectProvider().apply(variableManager));
+        }
+        return Optional.empty();
+    }
+
+    private Optional<TreeDescription> getTreeDescription(IEditingContext editingContext, Tree tree) {
+        return this.representationDescriptionSearchService
+        .findAll(editingContext).values().stream()
+        .filter(TreeDescription.class::isInstance)
+        .map(TreeDescription.class::cast)
+        .filter(td -> Objects.equals(td.getId(), tree.getDescriptionId()))
+        .findFirst();
```

Comment:
Can be removed and replaced inline with a simple call to `IRepresentationDescriptionSearchService#findById` instead

Reply:
Done